app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -0,0 +1,49 @@
+# Serve the pillar information
+class InternalApi::V1::PillarsController < InternalApiController
+  def show
+    pillar = Hash[Pillar.pluck :pillar, :value]
+    # rubocop:disable Style/AlignHash
+    ok content: {
```

Comment:
this structure could be derived from `Pillar.all_pillars` so we wouldnt have to duplicate and maintain this rather big structure 2 times, which is error prone:

```ruby
ok content: {}.tap do |hash|
  Hash[Pillar.pluck :pillar, :value].each do |key, value|
    hash.deep_merge! key.split(":").reverse.inject(value) { |a, n| { n => a } }
  end
end.deep_symbolize_keys
```

this however doesnt include nil values, in my case e.g. `proxy[:http]` will not be rendered. i just hacked it quickly on the console...

Reply:
Yeah, I prefer to play safe at the moment with this, missing entries might lead to unexpected problems if the salt code doesn't have defaults...